b2luigi/core/task.py

```diff
@@ -76,6 +74,20 @@ def _transform_input(input_generator, key=None):
             return file_paths[key]
         return file_paths
 
+    def get_all_input_file_names(self):
+        """
+        Return all file paths required by this task.
+
+        Example:
+            class TheSuperFancyTask(b2luigi.Task):
+                def dry_run(self):
+                    for name in self.get_all_output_file_names():
+                        print(f"\t\toutput:\t{name}")
+        """
+        for file_name_key, file_names in self.get_input_file_names().items():
```

Comment:
Still a bit complicated :-)
There is `dictionary.values()`, which returns an iterable over the values. And you can do a `yield from iterable` instead of a for ... yield.
So your function will only be the single line `yield from self.get_input_file_names().values()` or just a single `return self.get_input_file_names().values()` (as it is an iterable already).

Reply:
oh man... sure true :P

Reply:
nahh... the problem is we have then a list in a list... we have at least iterate once through the values, or is `self.get_input_file_names().values()[0]` in all cases sufficient?

Reply:
unfortunately i was not able to find an elegant way to get rid of the dict_value object

Reply:
transfering the `dict_value` object in a list will lead to:
```
[['/path/to/a', '/path/to/a']]
```
or if file names are not equal
```
[['/path/to/a'], ['/path/to/b']]
```

Reply:
Right, because each value itself is a list. You are correct, I missed that.
Two possibilities I see here. Either, use `yield from`:
```python
for file_names in self.get_input_file_names().values():
    yield from file_names
```

or (shorter):

```python
return sum(self.get_input_file_names().values(), [])
```

whatever you prefer.

Reply:
I have just seen your solution. Also completely fine with me.